### ISSUE TITLE
Running a script from the php cli

### ISSUE THREAD

**Issue by brunolustosa** [qtranslate/qtranslate-xt#1197]
Hello.
Tried to create a simple script for importing some stuff into WordPress.
To get WP functions available to my script, I included 'wp-load.php', as I usually do all the time.
However, importing wp-load.php with qtranslate-xt installed results in the script dying there.
I debugged it a little, and saw that is tries to redirect the user somewhere, by checking HTTP_HOST and REQUEST_URI in $_SERVER. This is in qtranslate_core.php, in function qtranxf_init_language(). Ideally, it should not check for these variables if running from the CLI, as they will not be available.
I simply created a $_SERVER, and added those two keys to it, and it worked, although with a few problems.
To make it work, I set HTTP_HOST to the hostname of the site I was working on, and set REQUEST_URI to '/xx/' (xx being the language code). Tried setting it to '/', but it didn't work.
The problem is, as I set REQUEST_URI to a specific language, most of the functions I called from WP passed data through filters. For example, when using get_field() (from ACF) or even get_post_meta() would return data translated to that language. To work around this, I used wpdb and did a direct query in the database, to get the whole string with the all the languages (with shortcodes).
To summarize, from my understanding, running from the CLI should:
- ignore $_SERVER variables, as they are not typically available;
- not depend on setting a default language by checking the REQUEST_URI. To get something in a specific language, I could use qtranxf_use() to translate.
I got it all solved in my script, and was able to do what I needed, but took me some time to figure out why it was dying when including wp-load.php.
Thanks,

**Comment by spleen1981** [comment]
It seems to be a very specific use case and non-standard use of wordpress to me.
Don't think this needs to be handled within QTX.
I guess standard way would be running any script of this kind from the WP admin interface (e.g. specific plugin, child theme, etc.), I can believe some hacking is needed otherwise.

**Comment by herrvigg** [comment]
Indeed it's for a specific use case. I can't tell yet if this is easily feasible or not. I leave it as a feature request for now.
